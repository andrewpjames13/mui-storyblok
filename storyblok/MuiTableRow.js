module.exports = {
  "name": "MuiTableRow",
  "display_name": null,
  "updated_at": "2020-05-06T14:54:16.580Z",
  "schema": {
    "rootClass": {
      "type": "options",
      "description": "stroyblok multiselect of css classes\n* Mui Override or extend the styles applied to the component.",
      "options": [
        {
          "value": "font-weight: bolder",
          "name": "font-weight"
        },
        {
          "value": "",
          "name": ""
        }
      ]
    },
    "hover": {
      "type": "boolean",
      "description": "  * mui prop: true | false\n   * If true, the table row will shade on hover."
    },
    "selected": {
      "type": "boolean",
      "description": "   * mui prop: true | false\n   * If true, the table row will have the selected shading.\n"
    },
    "content": {
      "type": "bloks",
      "restrict_components": true,
      "component_whitelist": [
        "MuiTableCell"
      ],
      "required": true
    }
  },
  "image": null,
  "preview_field": null,
  "is_root": false,
  "preview_tmpl": null,
  "is_nestable": true,
  "all_presets": [],
  "preset_id": null,
  "real_name": "MuiTableRow",
  "component_group_uuid": null
}