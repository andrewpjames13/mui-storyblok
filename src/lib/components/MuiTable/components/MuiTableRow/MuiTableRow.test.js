import React from 'react';
import { shallow } from 'enzyme';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import MuiTableRow from './MuiTableRow';

function setup() {
  const props = {
    hover: false,
    selected: false,
    content: [{
      size: 'medium',
      align: 'center',
      content: [{
        align: 'inherit',
        color: 'initial',
        width: '100%',
        height: '100%',
        content: [{
          text: 'Single table row',
          component: 'MuiText',
        }],
        variant: 'body1',
        component: 'MuiTypography',
      }],
      padding: 'default',
      component: 'MuiTableCell',
      sortDirection: 'asc',
    }],
  };
  const comp = shallow(<MuiTableRow {...props} />);
  return { comp, props };
}

describe('<MuiTableRow />', () => {
  test('snapshot', () => {
    const { props } = setup();
    const tree = renderer.create(
      <MemoryRouter>
        <MuiTableRow {...props} />
      </MemoryRouter>,
    );
    expect(tree).toMatchSnapshot();
  });

  it('should render MuiTable Row', () => {
    const { comp } = setup();
    expect(comp).toBeDefined();
  });
});
