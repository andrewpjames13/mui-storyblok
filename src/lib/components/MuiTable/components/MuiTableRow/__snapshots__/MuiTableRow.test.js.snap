// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MuiTableRow /> snapshot 1`] = `
<tr
  className="MuiTableRow-root makeStyles-root-2"
>
  <td
    aria-sort="ascending"
    className="MuiTableCell-root makeStyles-root-8 MuiTableCell-alignCenter"
  >
    <p
      className="MuiTypography-root makeStyles-root-21 MuiTypography-body1"
    >
      Single table row
    </p>
  </td>
</tr>
`;
